requirements-dev.txt

```
@@ -18,7 +18,7 @@ seaborn>=0.9.0
 reno>=3.4.0
 Sphinx>=3.0.0
 qiskit-sphinx-theme>=1.6
-sphinx-autodoc-typehints>=1.18
+sphinx-autodoc-typehints==1.23.1  # revert to >=1.18 once the incompatibilities in >1.23.1 are fixed
```

```suggestion
sphinx-autodoc-typehints==1.21.1  # revert to >=1.18 once the incompatibilities in >1.23.1 are fixed
```
It should be `1.21.1` not `1.23.1`. See https://github.com/Qiskit/qiskit-terra/pull/9383#issuecomment-1386778745

---

done in 8a941c9